frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/data-resiliency/data-resiliency.tsx

```diff
@@ -77,9 +77,9 @@ const DataResiliency: React.FC<DashboardItemProps> = ({
   const totalPg = getCapacityStats(totalPGRaw);
   const cleanAndActivePg = getCapacityStats(cleanAndActivePGRaw);
 
-  let progressPercentage;
+  let progressPercentage: number;
   if (totalPg && cleanAndActivePg) {
-    progressPercentage = ((Number(cleanAndActivePg) / Number(totalPg)) * 100).toFixed(1);
+    progressPercentage = Number(((Number(cleanAndActivePg) / Number(totalPg)) * 100).toFixed(1));
```

Comment:
it seems like there's lot of number parsing/NaN checks. How about using doing
```suggestion
    progressPercentage = ((Number(cleanAndActivePg) / Number(totalPg)) * 100).toFixed(1);
    if(!isFinite(progressPercentage)){
      progressPercentage = 0;
    }
```

and then later when passing data to `DataResiliencyStatusBody` we could be more transparent and add `error` property which would indicate when to show `not available` state (ie when cleanAndActivePg and totalPg are not numbers).

Reply:
ack